libs/product/testing/src/factories/product.factory.ts

```diff
@@ -17,6 +17,7 @@ export class MockProduct implements DaffProduct {
 	type = DaffProductTypeEnum.Simple;
 	id = faker.datatype.uuid();
 	url = `/${faker.random.word()}.html`;
+	canonicalUrl = `/${faker.random.word()}.html`;
```

Comment:
For correctness, give this a full URL with protocol and domain.

Reply:
changed